### DISCUSSION TITLE
How come my page gets rendered before getStaticProps?

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/next.js#19979] [Help]
I've been away from Next.js for a bit and I'm finding this a bit confusing. It seems like my pages get rendered before `getStaticProps` is called, which causes quite a lot of problems since a lot of `context` values are undefined on first render.

On a simple page composed by:

```Javascript
export default function ServerPage(props) {
  console.log("props are", props.params);
  return Test;
}

export async function getStaticProps(context) {
  console.log("params are", context.params);
  return {
    props: { params: context.params.id },
  };
}
```

My console logs will be:
```
props are undefined
params are { id: '2' }
props are 2
```

Is this what's supposed to happen? Surely the my page shouldn't render if all the props passed to it are undefined? How do I handle hooks that can't be conditionally rendered if I still want to mantain SSR and static exporting?

**Comment by leo-petrucci** [comment]
bump

**Comment by jamesmosier** [comment]
Do you have an example repository you could share? I'm not able to reproduce with your code sample provided.

**Reply by leo-petrucci** [reply]
@jamesmosier I ended up asking on reddit and apparently the problem was that I wasn't specifying the path in `getStaticPaths` and landing on the fallback page.

It's unfortunate the fallback page is still rendered server-side as it makes it quite hard to develop fallback pages but that was the problem!
